lib/new_relic/agent/method_tracer_helpers.rb

```diff
@@ -26,12 +27,58 @@ def trace_execution_scoped(metric_names, options = NewRelic::EMPTY_HASH) # THREA
           segment.record_metrics = false
         end
 
+        unless !options.key?(:code_information) || options[:code_information].nil? || options[:code_information].empty?
+          segment.code_information = options[:code_information]
+        end
+
         begin
           Tracer.capture_segment_error(segment) { yield }
         ensure
           segment.finish if segment
         end
       end
+
+      def class_info(object)
+        name = object.name
+
+        return [name, true] if name
+        return [Regexp.last_match(1), false] if object.to_s =~ /^#<Class:(.*)>$/
+
+        ::NewRelic::Agent.logger.error("Unable to determine name for what was expected to be a singleton class, #{object}")
```

Comment:
I wonder if this might be a nice area to add a Supportability metric to capture data on how often this occurs?

Reply:
This is a terrific idea! This code has been updated again with #1085, but once it settles down I'd like to layer in support for that metric for sure.

Reply:
@kaylareopelle can you let me know if these lines accomplish what you wanted to see happen?

[method_tracer_helpers.rb#75](https://github.com/newrelic/newrelic-ruby-agent/pull/1069/files#diff-7e20a306b4d12a8e387870b566cf31215725ae5abad2e26d592b15521da78bb4R75)

[method_tracer_helpers.rb#84](https://github.com/newrelic/newrelic-ruby-agent/pull/1069/files#diff-7e20a306b4d12a8e387870b566cf31215725ae5abad2e26d592b15521da78bb4R84)

Reply:
Yes! They look great. My one recommendation would be to add `Ruby` in the metric name so that we can see these values in Angler (in case any other agents have the same metric). The language was the second-to-last value in all the Supportability metrics we created for Logging.

Reply:
Updated the metric name and dropped the success condition reporting with 5cfe616